android/src/com/mapswithme/maps/MwmActivity.java

```diff
@@ -2720,4 +2723,31 @@ public void onGlobalLayout()
                               : UiUtils.getStatusBarHeight(getApplicationContext()));
     }
   }
+
+  @Override                                                                                                                                        
+    public boolean onKeyUp(int keyCode, KeyEvent event) {                                                                                            
+      switch (keyCode) {                                                                                                                             
+        case KeyEvent.KEYCODE_DPAD_DOWN:                                                                                                             
+          Statistics.INSTANCE.trackEvent(Statistics.EventName.ZOOM_OUT);                                                                             
+          AlohaHelper.logClick(AlohaHelper.ZOOM_OUT);                                                                                                
+          MapFragment.nativeScaleMinus();                                                                                                            
+          return true;                                                                                                                               
+        case KeyEvent.KEYCODE_DPAD_UP:                                                                                                               
+          Statistics.INSTANCE.trackEvent(Statistics.EventName.ZOOM_IN);                                                                              
+          AlohaHelper.logClick(AlohaHelper.ZOOM_IN);                                                                                                 
+          MapFragment.nativeScalePlus();                                                                                                             
+          return true;                                                                                                                               
+        case KeyEvent.KEYCODE_ESCAPE:                                                                                                                
+          if (Config.isWunderLINQEnabled()) {                                                                                                        
+            //Go Back To WunderLINQ                                                                                                                  
+            String callingApp = "wunderlinq://datagrid";                                                                                             
```

Comment:
Could we avoid the hardcoded uri of your application in our code? Perhaps, the another application also wants to intercept the KEYCODE_ESCAPE and run their application...My colleague @beloal propose to use the intent/deeplink extra parameter that would consist the uri scheme on which we should come back by KEYCODE_ESCAPE. What do you think?

Reply:
I agree, if we move to this, maybe I should drop the "Enable WunderLINQ" preference?  Thats the only reason I use it.

Reply:
> I agree, if we move to this, maybe I should drop the "Enable WunderLINQ" preference? Thats the only reason I use it.

Yes, if we make parameter for "back" app we don't need in a preference and it may be safely deleted.

Reply:
I now capture an argument &backurl= and use that instead of hardcoding to wunderlinq://